File: tests/src/com.google.protobuf/protobuf-javalite/3.25.3/src/test/java/com_google_protobuf/protobuf_javalite/InternalTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_google_protobuf.protobuf_javalite;

import static org.assertj.core.api.Assertions.assertThat;

import com.google.protobuf.Internal;
import com.google.protobuf.MessageLite;
import com.google.protobuf.StringValue;
import org.junit.jupiter.api.Test;

public class InternalTest {
    @Test
    void getDefaultInstanceReturnsGeneratedMessageDefaultInstance() {
        StringValue defaultInstance = Internal.getDefaultInstance(StringValue.class);

        assertThat(defaultInstance)
                .isSameAs(StringValue.getDefaultInstance())
                .isInstanceOf(MessageLite.class);
    }
}

File: tests/src/com.typesafe.akka/akka-protobuf-v3_3/2.6.21/src/test/scala/com_typesafe_akka/akka_protobuf_v3_3/InternalTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_typesafe_akka.akka_protobuf_v3_3

import akka.protobufv3.internal.DescriptorProtos
import akka.protobufv3.internal.Internal
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Test

class InternalTest {
  @Test
  def getDefaultInstanceResolvesGeneratedMessageDefaultAccessor(): Unit = {
    val defaultInstance: DescriptorProtos.FileDescriptorProto = Internal.getDefaultInstance(
      classOf[DescriptorProtos.FileDescriptorProto]
    )

    assertSame(DescriptorProtos.FileDescriptorProto.getDefaultInstance, defaultInstance)
    assertEquals("", defaultInstance.getName)
    assertEquals(0, defaultInstance.getDependencyCount)
  }
}
